src/libstd/sys/windows/process.rs

```diff
@@ -141,12 +137,32 @@ impl Command {
                                    .with_extension(env::consts::EXE_EXTENSION);
                     if fs::metadata(&path).is_ok() {
                         return Some(path.into_os_string())
+                    } else {
+                        // Windows relies on path extensions to resolve commands.
+                        // Path extensions are found in the PATHEXT environment variable.
+                        if let Some(exts) = env_lookup(self.env.as_ref(), "PATHEXT") {
+                            for ext in split_paths(&exts) {
+                                let ext_str = pathext.to_str().unwrap().trim_matches('.');
```

Comment:
Sounds good to me! What about using `to_string_lossy()` instead? Something like this:
```rust
                            for ext in split_paths(&exts) {
                                let ext_str = ext.to_string_lossy();
                                let path = path.with_extension(
                                                ext_str.trim_matches('.')
                                );
                                if fs::metadata(&path).is_ok() {
                                    return Some(path.into_os_string())
                                }
                            }

```